### DISCUSSION TITLE
Turborepo + Preconstruct Entrypoints

### DISCUSSION THREAD

**Discussion by hmans** [vercel/turborepo#2245] [Help] [answered]
I'm using Turborepo together with Preconstruct.

Preconstruct has an [opinionated way of implementing additional entrypoints for packages](https://preconstruct.tools/guides/adding-a-second-entrypoint) that involves creating a tiny package directory _next to_ `dist` (not inside it) with a minimal `package.json` inside. This allows the consuming project to eg. `import "package/foo"` without forcing the package to commit to the new `exports` syntax (that, from what I've heard and seen, has compatibility issues with certain popular bundlers and environments.)

This is all fine and good, except now of course I need to make sure that turborepo recognizes these additional directories as part of a package's output. I basically have two questions:

1) In a monorepo with many packages, some of them having these extra entrypoints, and not all of these entrypoints sharing the same name, how do I configure turborepo correctly to include these in its view of package output?

2) Is there any way to maybe infer this from each package's configuration, eg. the `files` array in `package.json`?

Any help would be highly appreciated. I've had to disable turborepo in [my monorepo](https://github.com/hmans/composer-suite) for now because it wasn't caching and playing back these entrypoints correctly, wreaking havoc with CI. :(

**Comment by anthonyshew** [comment, marked as answer]
Setting up some `inputs` globs for the code dependencies outside of your package should help here: https://turbo.build/repo/docs/reference/configuration#inputs
